### PR TITLE
Tests: Fix t.Parallel() errors in netdeploy package

### PR DESCRIPTION
**Pull request by michaeldiamant**
Enables https://github.com/kunwardeep/paralleltest on `netdeploy` by fixing linter warnings.  Incrementally moves the ball towards greater unit test parallelization, which reduces local + CI test durations.

I vetted for flakiness by running:
```
go test -race -count 10 ./netdeploy/...
```

Notes:
* The PR does _not_ intend to rework all sequential tests into parallel tests.  My objective is to enable already available parallelism throughout go-algorand.  And then make a pass through remaining sequential tests to see if/where more parallelism exists.

### CODE REVIEW

File: netdeploy/remote/deployedNetwork_test.go [hunk 1]
```diff
@@ -87,7 +88,7 @@ func TestCreateSignedTxBasic(t *testing.T) {
 	}
 }
 
-func TestCreateSignedTxAssets(t *testing.T) {
+func TestCreateSignedTxAssets(t *testing.T) { //nolint:paralleltest // Not parallel because it modifies config.Consensus
```
[the remark this answers is not in the document]
Comment (jdtzmn) [reply netdeploy/remote/deployedNetwork_test.go:91]: Is it not mutated on line 108? Assuming it is mutated on line 108, is there supposed to be a cleanup step that's missing?

[the remark this answers is not in the document]
Comment (jdtzmn) [reply netdeploy/remote/deployedNetwork_test.go:91]: Ah yes - you're correct. I had forgotten that the values of the map are structs.

[the remark this answers is not in the document]
Comment (jdtzmn) [reply netdeploy/remote/deployedNetwork_test.go:91]: I tried using `t.Parallel()` on these tests and they all passed. Empirically, I've found that tests which read config.Consensus don't have any issues being parallelized when other tests containing Consensus mutations are not parallelized. I've tried looking for a more concrete answer to this online to no avail. @cce do you know any more about how parallel vs. sequential tests are sandboxed? Perhaps they are run one after another?

[the remark this answers is not in the document]
Comment (jdtzmn) [reply netdeploy/remote/deployedNetwork_test.go:91]: I can confirm that I am able to add `t.Parallel()` to these tests locally and that the test cases pass even with `-count=40` and above.

**Review by jdtzmn — DISMISSED** [review]
Changes look good! I went ahead and parallelized the two tests that appear to mutate `config.Consensus` but do not, as @cce pointed out. I additionally parallelized test files in the `netdeploy` package that were not caught by the golangci regex.

**Review by algochoi — APPROVED** [review]
Rebased the changes off of `master`, and it passes the following commands: `go test -race -count 10 ./netdeploy/...` and `golangci-lint run --disable-all -E paralleltest netdeploy/...`

**Review by algorandskiy — APPROVED** [review]

**Pull request merged**